### DISCUSSION TITLE
Allow route wildcards for entry routes for multi-page entries

### DISCUSSION THREAD

**Discussion by ldstevens** [craftcms/cms#8387] [Ideas]
This is an idea to simplify working with multi-page entries in Craft easier. 

Background: I'm working on a review site. Each review is contained in a single entry. On the front end each review is spread over multiple pages, and each page is represented by one or more fields. 

Let's say the routes for a given review look like this:
/reviews/cool-product/ (intro wysiwyg field)
/reviews/cool-product/specs (specs wysiwyg field)
/reviews/cool-product/photos (list of assets)
/reviews/cool-product/conclusion (rating field + conclusion wysiwyg field)

The site also has different sections with different entry types, including different review types. Therefore I use entry routes with /sectionname/{slug} that point to a single entry handler template, which then fetches the appropriate template for a given entry type.

The template is simply:
```
{% include '_entryTypes/' ~ entry.type %}
```

This works for /news/{slug} and /review/{slug} but breaks down for multi-page reviews, as /review/{slug}/somepage throws a 404 (as expected). 

To work around this I use dynamic routes as well. I have a route /review/{slug}/{*} that catches /review/{slug}/somepage, sends that request to a review handler template, which fetches the `entry` object manually (based on the `slug`), and includes the same entry type template as the entry route does:
```
{% set entries = craft.entries.slug(slug) %}

{% for entry in entries %}	
{% include '_entryTypes/' ~ entry.type %}
{% endfor %}
```

(In the review entry template I use segment conditionals to determine what field should be displayed on which page, and if none match then exit 404.)

IMHO this could be simplified by having the wildcard route available in entry routes. That way the `entry` object would be available based on the `slug` like normal entry routes, it would just be up to you to handle 404 and other routing in your templates. You could then create sub-pages as you please by managing segments. 

Hope that makes sense! :)

**Comment by ldstevens** [comment]
Just wanted to add a couple of thoughts for this one.

Multi-page entries throw up a couple of other issues beyond routes, specifically:
- Search results, and surfacing the correct "page" for a given entry in search results. 
- Preview, and navigating between pages in the preview iframe. I have a couple of workarounds in mind for this (using ajax nav for preview requests, or just showing everything on one page for reviews).

There's probably more that could go into multi-page entries (e.g. are they set natively with a title and slug? Can you loop through them to generate table of contents? etc etc). Happy to share more if/when you're interested in tackling this one. Cheers :)

**Comment by masiorama** [comment]
This is interesting. Did you manage to dig more into the matter? Thanks in advance.
